Hooks: Fix exec_command usage in PySide2.QtQuick

tobinjones opened the pull request:
As requested, this pull request fixes #3689.

Now imports `exec_command_stdout`.

The qmldir is also now checked correctly (the logic was backwards).

bjones1 commented:
This is a good start! Thanks!

Would you:

1. Rebase your code to the latest commits? Changes there will affect the next step.
2. Modify `tests/requirements-libraries.txt` to add PySide2 to the list of ? It looks like it's installable from PyPI, and there are already tests in `tests/function/test_libraries.py`.

bjones1 commented:
1. Only support current PySide2. PyInstaller, for the most part, doesn't try to support old versions. (A major exception to this PyQt4/PyQt5, since these are so widely used).
2. Include it in this PR.

IMHO, the whole PySide2 approach is broken -- it shouldn't depend on qmake at all. Instead, I'd recommend adding `pyqt5_library_info = Qt5LibraryInfo('PySide2')` [here](https://github.com/pyinstaller/pyinstaller/blob/develop/PyInstaller/utils/hooks/qt.py#L81), then changing all other areas of the code to follow the same approach as PyQt5. However, this is a much larger project and will take more time to implement.

tobinjones commented:
Okay, so I've:

- Rebased latest commits.
- Added `PySide2` to `tests/requirements-libraries.txt`.
- Replaced all uses of the `qmake` binary with queries to a `Qt5LibraryInfo` instance.

Works for me now!

bjones1 commented:
Great progress! It looks like there are two test failures:

1. PySide2 isn't available as a wheel on Python 3.4. Add something like `PySide==5.11.1; python_version != '3.4'`.
2. QWebEngine failures on Linux Python 2.7 and Windows Python. Any ideas on fixing these? Do they fail on your PC?

tobinjones commented:
1. Fixed the library requirements to reflect the lack of a wheel for python 3.4.
2. QWebEngine failures were due to things being moved around in the PySide2 module. The test works for me on my computer now (Win-64, Python3.7)

bjones1 commented:
Oops -- a typo from me -- it should be `PySide2==5.11.1; python_version != '3.4'`. Would you fix this? Also, please fix long lines per the lint complaints.

htgoebel commented:
Thanks for this pull-request. @bjones1 Is it ready to be merged?

Regarding the commit-message. Please use present tense also in the body of the commit-message. And since this commit contains (as far as I can see) three parts, I suggest using bullet points, like this

```
* Add function 'utils/hooks/pyside2_library_info', providing QLibraryInfo in
  the 'PySide2' namespace. This copies 'pyqt5_library_info'.

* Replace all use of the 'qmake' binary by the PySide2 hooks
  with calls to pyside2_library_info. This is tidier, and also 'qmake' is
  no longer supplied by the PySide2 module.

* Move the Qt path from a 'qt' subdirectory to the top level of the
  PySide2 module, so references to these were changed appropriately.
```

bjones1 commented:
@htgoebel, yes, this is ready.

tobinjones commented:
Hi, if you wait until Monday to merge, I’ll have a version of this that is better at dealing with the PySide wheels which are laid out quite inconsistently.

tobinjones commented:
Hi, if you wait until Monday to merge, I’ll have a version of this that is better at dealing with the PySide wheels which are laid out quite inconsistently.

htgoebel commented:
@tobinjones pinging

kefniark commented:
I was struggling with the `ImportError: cannot import name 'exec_command'` for hours on both windows & osx.
and thanks to this PR, I finally managed to fix the problem 👍

htgoebel commented:
@tobinjones Are you going to update and rebase this pull-request any time soon?

You also need to submit a [changelog entry](https://pyinstaller.readthedocs.io/en/latest/development/changelog-entries.html) so our users can learn about your change. (This is new since a few weeks now.)

DuaneKaufman commented:
@tobinjones  (@htgoebel) Are you going to release this any time soon?
I have a PySide2 app I am trying to build and this problem has me stymied.

DuaneKaufman commented:
Dear Bryan,
This is really unfortunate. Is there any way to contact the person originally working on this, as I thought they had nearly completed the fix. For someone else to come in and start over would be quite a task.
It doesn't look good that Pyinstaller cannot package a Qt app, as Qt use is quite popular.
Sincerely,Duane
----- Bryan A. Jones <notifications@github.com> wrote:
>I need a volunteer who can finish this up. It's fairly close. I'm closing this until someone steps forward.
> 
>—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or mute the thread.
>

bjones1 commented:
Not that I know of, other than using the system here. We're all volunteers -- Tobin has probably moved to another project.
